File: app/src/main/java/com/attendance/login/LoginContract.java
package com.attendance.login;

import com.attendance.BasePresenter;
import com.attendance.BaseView;

/**
 * Created by peiqin on 3/4/2017.
 */

public interface LoginContract {

    interface View extends BaseView<Presenter> {

        void showTip(String tip);

        void showProgress(String msg);

        void cancelProgress();

        void startMainActivity();

    }

    interface Presenter extends BasePresenter {

        void login(String username, String password, boolean isTeacher, boolean isRemPassword);

        void writeData();

    }

}

File: app/src/main/kotlin/com.attendance/contract/LoginContract.kt
package com.attendance.contract

import com.attendance.BasePresenter
import com.attendance.BaseView

/**
 * Created by peiqin on 3/4/2017.
 */

interface LoginContract {

    interface View : BaseView<Presenter> {

        fun showTip(tip: String)

        fun showProgress(msg: String)

        fun cancelProgress()

        fun startMainActivity()

    }

    interface Presenter : BasePresenter {

        fun login(username: String, password: String, isTeacher: Boolean, isRemPassword: Boolean)

        fun writeData()

    }

}
